lib/ecto/query/builder/dynamic.ex

```diff
@@ -40,14 +40,14 @@ defmodule Ecto.Query.Builder.Dynamic do
   list is not reversed. This is useful when the dynamic expression
   is given in the middle of an expression.
   """
-  def partially_expand(kind, query, %{binding: binding} = dynamic, params, count) do
-    {expr, {_binding, params, subqueries, count}} = expand(query, dynamic, {binding, params, [], count})
+  def partially_expand(kind, query, %{binding: binding} = dynamic, params, subqueries, count) do
+    {expr, {_binding, params, subqueries, count}} = expand(query, dynamic, {binding, params, subqueries, count})
 
-    if subqueries != [] do
+    if subqueries != [] and kind != :select do
```

Comment:
Maybe we should two partially_expand. One that accepts subqueries (and therefore returns subqueries) and another one without. I think I prefer this one.

Thoughts?

Reply:
That was my initial implementation. I was on the fence, but decided to expose the `subqueries` to all callers, in case we want to support subqueries for them in the future.

Looking at it again, I think they are actually structurally different - ordering, grouping or updating by a subquery doesn't really seem to make sense. So having two `partially_expand` seems to make more sense.

I've implemented the change in [583795f](https://github.com/elixir-ecto/ecto/pull/3927/commits/583795ff2c3b46847d41a225d2ebb25f40bef427). Not sure about the args list - I removed `kind` as it's not needed anymore, so it matches only on the position of the `%{binding: binding}`, which is not ideal but works. Open for suggestions!